src/services/redis/redis-broker.client.ts

```diff
@@ -0,0 +1,50 @@
+import redis, { createClient } from 'redis';
+
+import { BrokerProxy } from '../../models/interfaces/broker-proxy.interface';
+
+interface RedisClients {
+  subscriber: redis.RedisClientType<any>;
+  publisher: redis.RedisClientType<any>;
+}
+
+interface RedisPublishPayload {
+  channel: string;
+  message: unknown;
+}
+
+export interface RedisSubscribePayload {
+  channel: string;
+  message: string;
+}
+
+export class RedisBrokerClient extends BrokerProxy<unknown> {
+  private client: RedisClients;
+
+  async connect(): Promise<void> {
+    return new Promise(async (resolve) => {
+      const subscriber = createClient({ url: this.broker });
+      const publisher = createClient({ url: this.broker });
+      await subscriber.connect();
+      await publisher.connect();
```

Comment:
Maybe we can connect both in one shot.
```suggestion
      await Promise.all([
        subscriber.connect(),
        publisher.connect(),
      ]);
```